src/tests/base_test.py

```diff
@@ -78,6 +82,31 @@ def update_parameterized(cls, request: pytest.FixtureRequest, config: BaseConfig
                 else:
                     raise AttributeError(f"No attribute name {fixture_name} in {config_type} object type")
 
+    @pytest.fixture
+    def k8s_minikube(self, request: pytest.FixtureRequest):
+        """k8 minikube wrapper client  object fixture passed to test function
+        This function fixture called by testcase, we initialize k8client
+        Storing configmap before test begins -> for later restoring
+        When test func done, check if configmap was changed and restore config.
+        :return:
+        """
+        log.debug(f"--- SETUP --- Creating k8s_minikube for test: {request.node.name}\n")
+        k8s_client_ = kubernetes.dynamic.DynamicClient(
+            kubernetes.client.api_client.ApiClient(
+                configuration=kubernetes.config.load_kube_config(),
+            ),
+        )
+        minikube = MinikubeController(k8s_client_)
+        # before test begins verify all developments are ready
+        if not minikube.verify_deployments_are_ready():
+            assert "Minikube deployments services not ready/active"
+        configmap_before = dict(minikube.configmap_data["data"])
```

Comment:
I'm not sure that is will do a deepcopy to the dict, maybe change `configmap_data` property` so that it will not be available from outside the class, only a deep copy of it.

Reply:
Yes , i missed this part , it will return a shallow copy and if someone will change it from the returned reference we will  get the same before and adter, i changed the property configmap_data to copy.deepcopy , Thanks !!!

Reply:
Hi , i think that that the configmap python code is protected from direct assignment
ResourceField
config_map_store[0]['data']['OS_IMAGES'] = 5
TypeError: 'ResourceField' object does not support item assignment

So if someone copy the configMap python we would not able to modify because all data fields are immutable (str and int) so no worries that someone snick into a list and add elements.....     the only way to hook it but i think its a rare case is by config_map_store[0]['data'].__dict__['OS_IMAGES']  .  looks like we do not need to change it . no real risk here